Corona-Warn-App/src/main/java/de/rki/coronawarnapp/appconfig/AppConfigSource.kt

```diff
@@ -0,0 +1,82 @@
+package de.rki.coronawarnapp.appconfig
+
+import de.rki.coronawarnapp.appconfig.download.AppConfigServer
+import de.rki.coronawarnapp.appconfig.download.AppConfigStorage
+import de.rki.coronawarnapp.appconfig.download.ApplicationConfigurationInvalidException
+import de.rki.coronawarnapp.appconfig.mapping.ConfigParser
+import de.rki.coronawarnapp.util.coroutine.DispatcherProvider
+import kotlinx.coroutines.withContext
+import timber.log.Timber
+import javax.inject.Inject
+import javax.inject.Singleton
+
+@Singleton
+class AppConfigSource @Inject constructor(
+    private val server: AppConfigServer,
+    private val storage: AppConfigStorage,
+    private val parser: ConfigParser,
+    private val dispatcherProvider: DispatcherProvider
+) {
+
+    suspend fun retrieveConfig(): ConfigData = withContext(dispatcherProvider.IO) {
+        Timber.v("retrieveConfig()")
+        val (serverBytes, serverError) = try {
+            server.downloadAppConfig() to null
+        } catch (e: Exception) {
+            Timber.tag(TAG).w(e, "Failed to download AppConfig from server .")
```

Comment:
rather an error (.e.) log?

Reply:
As this would also trigger when no internet is available, it's more of a warning, isn't it? An error will be thrown and logged if then also no fallback config is available.